v03_pipeline/lib/reference_data/compare_globals.py

```diff
@@ -15,131 +19,125 @@
 logger = logging.getLogger(__name__)
 
 
+def struct_to_dict(struct: hl.Struct) -> dict:
```

Comment:
does something like `dict(hl.Struct(a=1, b=2))` work here?

Reply:
it does but if it's nested like this, 
```
g = hl.Struct(
                versions=hl.Struct(),
                paths=hl.Struct(
                    a='gs://a_path_OLD.ht',
                ),
                enums=hl.Struct(),
            )
print(dict(g))
```
prints {'versions': Struct(), 'paths': Struct(a='gs://a_path_OLD.ht'), 'enums': Struct()}, 

I can simplify the function a bit by converting the struct into a dict initially:
```
def struct_to_dict(struct: hl.Struct) -> dict:
    struct = dict(struct)
    for field in struct:
        if isinstance(struct[field], hl.Struct):
            struct[field] = struct_to_dict(struct[field])
    return struct
```

Reply:
do you think it's just unnecessary to turn structs into the dicts? I think my reasoning for this was that it was easier for me to think about equality checks with just simple python data structures as opposed to hail structs

Reply:
I'm not sure!  It looks like hail's types work just fine with `__eq__`:

```
In [5]: hl.Struct(a=1,b={'d': 2}) == hl.Struct(a=1,b={'d': 1})
Out[5]: False

In [6]: hl.Struct(a=1,b={'d': 1}) == hl.Struct(a=1,b={'d': 1})
Out[6]: True
```